### ISSUE TITLE
Match with multiline string breaks compilation

### ISSUE THREAD

**Issue by stefan-schweiger** [fsprojects/fantomas#2584]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMiAeAlgM5YEB2A5gMKwCuZWAOmRBOQA61YCCzrYHLgCE%2BrALyiIAWwCGWaAAswACkE8ANALKcsQgJRgA7gSwLJYAD5hGkGzfTlEARmYOyiAEyvHAZmZ3bWxZ%2BVk01ITAAWgA%2Ba0gASW1hCUgwAGotHRFgiCsAfSjYgDMZAnRjUzjEgDcZB3gwAEdaRAAnAE8bSHVwKCJsUkoiVDAAbV6IUBCoABIsdvZkNEhyLG6J1kgZ2vQWkfHpiAAGHsOoRKQGAGUCAC9kU8OAFg2IAF0NgF9HydfZ%2BcWIxWDHW0y2Oz2aAOhycPxCkAAsjI8AAZRwoxCUUygmEeI5-D78b4bKbwuYLJZQTHwADyRTR7iu80wOP44LqkLGfwgHjhbIAomRafTHKz4egipACV84aS2eTAcsAEawWDoMUQdm7SnQ6Y%2BPmbRL9VpYABi5DqADlEIY3A9uWAsK0WtKibK-lsAZTICq1RrZhCdQ6ngbNVd2DJoIghIgirBWogAAoyVoyKTYNr%2BiBOl3TQmsYn8OWbBXe33qg1azm6kIAVlDUHDkejsfjiBRsEMbWgMn6NVgPdIsDIWcdzsQroL7rBpaB5f9VaDZwAbA3IE2ozG4wmAKrsRatHt9sjVAdyAjD0cldD9ScQQusYua2fK1UVj3bDlLw4AdjXG5bbdECodBeyIGgyBIZ1oCweMrzqW88xlEkPy9Oc3wXT9tX2B0AA5-wjTdWwTBFECkJVMwba9EJCfN72nMk0NfP1Kyw6sHQATgI5stzbGh0EvKiEInJC3RQmcmKgedWMDHCzicE4HXXQjEG4IosDaGgpFkUccxE2jkKLVCKXQliP1kqEHScWElIA3iEyuMiCDgASRyEm99P4OiwAfX4JJM5j3xnCyuXk3lbJUtSNNaRypGctVBIdPS7x8hj5Ukn0MJkr85JhfUIubbhWjoIUABFEAcOLot08cUt8sAn3%2BAKoFWTCQprfgnBDJSkTweIigAFQUTErgUeMsAAdQIeBsQbfFRKncTGOa4E1my7DLPk%2BseuRfqhsxfkPNG8appmhRR2XeaDLEoz-MVFqQXW9j5NXHa%2BrIIoCDwGkDzkeN%2BTwdgEyIIgLzcvCrq8wzH2M%2B7VranLNphP83oAJUQOBWngU7ZuDSHWG8%2BrGs9FbWqe79picfC0Yx%2BN4EtWhyLaOl4g0qRhgbJw6rSksMqkWh0FIO1TXjWQsGigaTPJ3LKa4pT0cx%2BAEQFoXHBF1oxeqtdFBTSNoryYwzqlBb6KW9LSce8zEdCw48TXXqitTdoaVaNESBx86G1w-H3mhvzlrhsmrY2m3pg8GyzkRZFHZkZ3XeILAGaZ1oWbZjmrO5s3eZW-nBbKNXRbkCWpeD57bfCyOY7jt2sGVvPhcL8XKKUnXUxgtoDem7FM9ugPvSD4LrY61gPHyyPeoANS-IRyHgcgKA90dvZ7mG7v7y3B5D4eeW68fkVNegYLBmehXnxeGyeH2wEJnnnwygeyXah0PG2ve8FK2AsAAcWwAGgcQEGYNz4QxXv7c2cNpKlwpiEDwr1x4q3zu4IQAloAAGshBt1QdgIgNIyBXDTMBNUjNwZnGop5AmfsGqwzLFlKBMsYEo0jtaW0jgYxYC7JiSWixyqfTICYMG3AhSkWTunM4yUTapSznfFakDN5lzDtTSuDgKBkAPmQI%2Bw4bgqLkLQBMA1YAXExFgc8iVSHCVAVQtepkgqPyHs-OWSjopkHPNUdsw4KDCIoq0Hh5B%2BHDlEYcMhFjiYvikrQuR0D%2BA%2BEUvAvODJ2xpiVPAGQIFYCg0oMwu08EPLBOodYhGW8HQ%2BAjocSAf82gEHTAwOoABpRAiB2CGIYIkDBMh1GewdEEiRRM8mBQKfIkIPgK6lOQW01B8SiBFRKvTABGklYILtH-YGoN-E1VzNdRavdwE0LMhE%2BhUSx4jJTPZRApViDQFaJUi0cydx8OHOVaAoFUxDhIYE8x3Tb5NQgeE2xhSzg%2BF3qU8plyqnGPQEyYqtAoK0HYEydoLJ3I0ShjdVefcgQP3lE-f5r9Sl1Iab1JOXi6SjLIOMxwATKZHCvjfKRXydk2MxXY-5cDSkQucrXBA9ozE5O6b0N4jwVhEFNKDEYQTPhAA)

#### Code

```fsharp
let existingCount
    inputA
    inputB
    =
    match (inputA, inputB) with
    | """line1
line2
line3
"""
        , inputB -> "InputB=" + inputB
    | _ -> failwith "Invalid query"
```
            

#### Result

```fsharp
let existingCount inputA inputB =
    match (inputA, inputB) with
    | """line1
line2
line3
""",
      inputB -> "InputB=" + inputB
    | _ -> failwith "Invalid query"

```
            
#### Problem description

When matching of two elements and one of them is a multiline string the original code works without issue. But the formatted code breaks compilation.

```
(6,3) (6, 4)Error10parse
Incomplete structured construct at or before this point in pattern matching. Expected '->' or other token.

(7,13) (7, 15)Error10parse
Unexpected symbol '->' in binding. Expected incomplete structured construct at or before this point or other token.
```

#### Extra information

- [x] The formatted result breaks my code.
- [x] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas main branch at 2022-10-14T16:59:37Z - 35f093ce1b8a19d9e9ade3da2c0f855923d8a106

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub>

**Comment by stefan-schweiger** [comment]
I found a pretty easy workaround for this issue when you put brackets around the match cases the code doesn't break:

```
let existingCount inputA inputB =
    match (inputA, inputB) with
    | ("""line1
line2
line3
""",
       inputB) -> "InputB=" + inputB
    | _ -> failwith "Invalid query"
```

So I guess maybe the formatter should just know that in such a case it needs to add brackets to the formatting.

**Comment by nojaf** [comment]
Have you considered extracting the multiline string into its own value?
Something like
```fsharp
[<Literal>]
let Lines = """line1
line2
line3
"""

let existingCount
    (inputA: string)
    (inputB: string)
    =
    match (inputA, inputB) with
    | Lines, inputB -> "InputB=" + inputB
    | _ -> failwith "Invalid query"
```

The code formatter should always be able to produce valid code, but sometimes a tiny refactor can solve the edge case at hand as well.

**Comment by stefan-schweiger** [comment]
The real code is somewhat more complex and has about 10 of those cases with additional `when` conditions and is used in testing to validate generated queries and the parameters with additional `when` conditions.

You are right that the code is not very good to begin with because the readability is low. But for our specific implementation separating the multiline text from the match makes it even harder to understand what's going on in the test cases.

The workaround with just adding brackets is fine for now for us as this code doesn't get updated very often and it's the only place were we handle it like this. But as you said a code formatter shouldn't break valid code :)

**Comment by nojaf** [comment]
Fix available in [v5.2.0-beta-001](https://www.nuget.org/packages/fantomas/5.2.0-beta-001)
